packages/api/test/e2e/api/promise-contract.spec.ts

```diff
@@ -2,70 +2,91 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { Address, Hash } from '@polkadot/types/interfaces';
-
 import fs from 'fs';
 import path from 'path';
-
 import { Abi } from '@polkadot/api-contract';
 import testingPairs from '@polkadot/keyring/testingPairs';
 import WsProvider from '@polkadot/rpc-provider/ws';
 
-import { SubmittableResult } from '../../../src';
-import ApiPromise from '../../../src/promise';
-import describeE2E from '../../util/describeE2E';
-
-import flipperAbi from '../../../../api-contract/test/contracts/flipper.json';
-const flipperCode = fs.readFileSync(path.join(__dirname, '../../../../api-contract/test/contracts//flipper-pruned.wasm')).toString('hex');
+import { Address, Hash } from '@polkadot/types/interfaces';
 
-describeE2E()('Promise e2e contracts', (wsUrl): void => {
+import { ApiPromise, SubmittableResult } from '../../../src';
+import { describeE2E } from '../../util';
+
+// Backwards compatibility:
+// A breaking change was introduced by substrate runtime spec version 97. https://github.com/paritytech/substrate/pull/2911/files
+// The change had to be implemented in ink! which changed the structure of the Wasm files.
+// The Polkadot JS API is only supporting substrate version greater or equal to spec_version 97.
+
+import incrementerAbi from '../../../../api-contract/test/contracts_97_/Incrementer.json'; // eslint-disable-line
+const incrementerCode = fs.readFileSync(path.join(__dirname, '../../../../api-contract/test/contracts_97_/incrementer-pruned.wasm')).toString('hex'); // eslint-disable-line
+
+describeE2E({
+  except: [
+    'docker-polkadot-master',
+    'docker-polkadot-alexander',
+    'docker-substrate-1.0',
+    'remote-polkadot-alexander',
+    'remote-substrate-1.0'
+  ]
+})('Promise e2e contracts', (wsUrl: string): void => {
   let address: Address;
   let codeHash: Hash;
   const keyring = testingPairs({ type: 'sr25519' });
   let api: ApiPromise;
+  let CREATION_FEE: number;
 
   beforeEach(async (done): Promise<void> => {
     api = await ApiPromise.create(new WsProvider(wsUrl));
+    CREATION_FEE = api.consts && api.consts.contracts
```

Comment:
Unneeded since we only support 2.x (as per exclusions)

Reply:
true, changed.